File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        minDel = b = 0
        for ch in s:
            if ch == 'b':
                b += 1
            else:
                minDel = min(minDel + 1, b)

        return minDel

File: 1653-minimum-deletions-to-make-string-balanced/1653-minimum-deletions-to-make-string-balanced.java
class Solution {
    public int minimumDeletions(String s) {
        int minDel = 0, b = 0;
        for(var ch : s.toCharArray()){
            if(ch == 'b')
                b++;
            else
                minDel = Math.min(minDel + 1, b);
        }

        return minDel;
    }
}